lib/doctest.js

```diff
@@ -241,14 +246,16 @@ function processLine(
       });
       output ($test);
     } else if (accum.state === INPUT || accum.state === LOG) {
-      accum.state = OUTPUT;
-      value = trimmedLine;
       $test = last (accum.tests);
-      $test[OUTPUT].push ({
-        channel: null,
-        value: value
-      });
-      output ($test);
+      if (!$test[OUTPUT].some (isStandardOutput)) {
```

Comment:
This if statement actually describes another state in our state machine implicitly. It would not have to be there if we had two separate states for logging: `SYNC_LOG` and `ASYNC_LOG`. While the state machine is in `SYNC_LOG`, it allows transitions to the `OUTPUT` state, but while it's in `ASYNC_LOG` it only allows transitions to itself or `CLOSED`.

It might be nicer to implement it like that.

Reply:
The valid state transitions would then be something like:

```
CLOSED -> CLOSED
CLOSED -> OPEN

OPEN -> INPUT

INPUT -> INPUT
INPUT -> OUTPUT
INPUT -> SYNC_LOG
INPUT -> ASYNC_LOG
INPUT -> CLOSED

OUTPUT -> OUTPUT
OUTPUT -> ASYNC_LOG
OUTPUT -> CLOSED

SYNC_LOG -> SYNC_LOG
SYNC_LOG -> OUTPUT
SYNC_LOG -> ASYNC_LOG
SYNC_LOG -> CLOSED

ASYNC_LOG -> ASYNC_LOG
ASYNC_LOG -> CLOSED
```